server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -299,35 +339,166 @@ private void updateToLoadAndDrop(List<Notice> notices, Map<String, LookupExtract
   private void takeSnapshot(Map<String, LookupExtractorFactoryContainer> lookupMap)
   {
     if (lookupSnapshotTaker != null) {
-      List<LookupBean> lookups = new ArrayList<>(lookupMap.size());
-      for (Map.Entry<String, LookupExtractorFactoryContainer> e : lookupMap.entrySet()) {
-        lookups.add(new LookupBean(e.getKey(), null, e.getValue()));
+      lookupSnapshotTaker.takeSnapshot(getLookupBeanList(lookupMap));
+    }
+  }
+
+  private void loadAllLookupsAndInitStateRef()
+  {
+    List<LookupBean> lookupBeanList = getLookupsListFromLookupConfig();
+    if (lookupBeanList != null) {
+      ImmutableMap.Builder<String, LookupExtractorFactoryContainer> builder = ImmutableMap.builder();
+      executorService = MoreExecutors.listeningDecorator(
+          Executors.newScheduledThreadPool(
+              lookupConfig.getNumLookupLoadingThreads(),
+              Execs.makeThreadFactory("LookupReferencesManager-Startup-%s")
+          )
+      );
+      try {
+        List<ListenableFuture<Map.Entry>> futures = startLookups(lookupBeanList);
+        final ListenableFuture<List<Map.Entry>> futureList = Futures.allAsList(futures);
+        try {
+          futureList.get()
+                    .stream()
+                    .filter(Objects::nonNull)
+                    .forEach(builder::put);
+
+          stateRef.set(new LookupUpdateState(builder.build(), ImmutableList.of(), ImmutableList.of()));
+        }
+        catch (InterruptedException | ExecutionException ex) {
+          futureList.cancel(true);
```

Comment:
Why need this call?

Reply:
Was attempting to cancel the loading of the pending lookups in case of an exception.